src/test/java/school/hei/haapi/validator/UserValidatorTest.java

```diff
@@ -0,0 +1,92 @@
+package school.hei.haapi.validator;
+
+import java.util.HashSet;
+import java.util.Set;
+import javax.validation.ConstraintViolation;
+import javax.validation.Path;
+import javax.validation.Validator;
+import javax.validation.metadata.ConstraintDescriptor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import school.hei.haapi.model.User;
+import school.hei.haapi.model.exception.BadRequestException;
+import school.hei.haapi.model.validator.UserValidator;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class UserValidatorTest {
+  Validator validator;
+  UserValidator userValidator;
+
+  @BeforeEach
+  void setUp() {
+    validator = mock(Validator.class);
+    userValidator = new UserValidator(validator);
+  }
+
+  @Test
+  void email_must_be_valid() {
```

Comment:
Useless. It's covered by IT. You do unit test for cases not covered by IT. If you covered everithing through IT (which is the case now), you don't need unit test